webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/services/EventListDisplayer.scala

```diff
@@ -102,12 +103,15 @@ class EventListDisplayer(
   private[this] val gridName = "eventLogsGrid"
 
   def display(refreshEvents:() => Box[Seq[EventLog]]) : NodeSeq  = {
+    val limit: Int = 500
     //common part between last events and interval
     def displayEvents(events: Box[Seq[EventLog]]) :JsCmd = {
       events match {
         case Full(events) =>
-          val lines = JsTableData(events.map(EventLogLine(_)).toList)
-            JsRaw(s"refreshTable('${gridName}',${lines.json.toJsCmd})")
+          val lines = JsTableData(events.map(EventLogLine(_)).toList.take(limit))
```

Comment:
You should order the events so you have the 500 newest elements, also you should not use take if size is < 500